### PR TITLE
Support for DateOnly, upgrade to Microsoft.Data.SqlClient, upgrade to NET6.0

### PR DESCRIPTION
**Pull request by asgerhallas**

### CODE REVIEW

File: src/HybridDb.Tests/Migrations/BuiltIn/HybridDb_1_x_x_to_2_x_x_Part1_Tests.cs [hunk 1]
```diff
@@ -20,9 +21,9 @@ public class HybridDb_1_x_x_to_2_x_x_Part1_Tests : HybridDbTests
         // To export data for use in these tests, use "Tasks > Generate Scripts" in SQL Management Studio.
         // Under "Advanced" select "Schema and data" for "Types of data to script"
 
-        // TODO: Test at et dokument, der loades uden configureret design via migration ikke fors�ges gemt i 
-        // databasen i en forkert tabel (Documents) og derved lader migrations g� i uendelig l�kke. 
-        // M�ske skal ManagedEntity have Design p� sig?
+        // TODO: Test at et dokument, der loades uden configureret design via migration ikke forsøges gemt i
```
Comment (peterbernth) [inline src/HybridDb.Tests/Migrations/BuiltIn/HybridDb_1_x_x_to_2_x_x_Part1_Tests.cs:24]: Is this comment outdated - thus it should be removed?

File: src/HybridDb/Config/ShortNameTypeMapper.cs [hunk 2]
```diff
@@ -1,35 +1,63 @@
...
+                var shortname = ShortNameByType(type);
+
+                if (!typesByShortName.TryGetValue(shortname, out var list))
+                {
+                    list = typesByShortName[shortname] = new List<Type>();
+                }
+
+                list.Add(type);
+
+                //if (!types.TryAdd(shortname, type))
```
Comment (peterbernth) [inline src/HybridDb/Config/ShortNameTypeMapper.cs:41]: outcommented - should be removed

File: src/HybridDb.Tests/DocumentSession_DateOnlyTests.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            using (var session = store.OpenSession())
+            {
+                var loaded = session
+                    .Query<EntityWith<DateOnly>>()
+                    .Single(x => x.Prop > new DateOnly(1899, 1, 1));
+
+                loaded.Prop.ShouldBe(new DateOnly(2022, 1, 2));
+            }
+        }
+    }
```
Comment (peterbernth) [inline src/HybridDb.Tests/DocumentSession_DateOnlyTests.cs:134]: Should we have a test for Store_Query_OrderBy?

File: src/HybridDb/JsonTypeHandler.cs [hunk 4]
```diff
@@ -10,8 +10,22 @@ public class JsonTypeHandler<T> : SqlMapper.TypeHandler<T>
...
+        public override void SetValue(IDbDataParameter parameter, T value) =>
+            throw new NotSupportedException("Logic for serializing is in DocumentDesigner class");
 
         public override T Parse(object value) => (T)serializer.Deserialize(value.ToString(), typeof(T));
     }
+
+    // There's yet no native support for DateOnly in Dapper, see https://github.com/DapperLib/Dapper/issues/1715
+    public class DateOnlyTypeHandler : SqlMapper.TypeHandler<DateOnly>
+    {
+        public override DateOnly Parse(object value) => DateOnly.FromDateTime((DateTime)value);
```
Comment (peterbernth) [inline src/HybridDb/JsonTypeHandler.cs:22]: Should we handle null?

File: src/HybridDb/Linq/Old/Parsers/LambdaParser.cs [hunk 5]
```diff
@@ -10,34 +10,51 @@ public class LambdaParser : ExpressionVisitor
...
         {
             if (expression.NodeType == ExpressionType.Quote)
+            {
                 Visit(expression.Operand);
+            }
 
             return expression;
         }
 
+        protected SqlExpression VisitPop(Expression expression)
```
Comment (peterbernth) [inline src/HybridDb/Linq/Old/Parsers/LambdaParser.cs:27]: Is this used?

**Review by peterbernth — CHANGES_REQUESTED** [review]

File: src/HybridDb/Linq/Old/Parsers/LambdaParser.cs (hunk 5 above)
Comment (asgerhallas) [reply src/HybridDb/Linq/Old/Parsers/LambdaParser.cs:27]: Nu er den :)

File: src/HybridDb/JsonTypeHandler.cs (hunk 4 above)
Comment (asgerhallas) [reply src/HybridDb/JsonTypeHandler.cs:22]: Now tested

**Pull request merged**